test/automated/ansible/provision.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
```

Comment:
Moved provision to bash script to add retries. Added 5 retries to `install-requirements` (that was randomly failing because of instance not reachable) and `macos-canaries` as they are idempotent